### ISSUE TITLE
Spike: Shared components with github submodules

### ISSUE THREAD

**Issue by Dianadec** [bcgov/reserve-rec-api#29]
#### Description:

Investigate if we can use submodules to share code between two projects.

#### Acceptance Criteria:
- [ ] decision of whether to use submodules or npm
- [ ] tested with backcountry reservation

#### Development Checklist:

- [ ] ...
- [ ] ...
- [ ] ...

#### Dependencies

- Blocked by
- Blocking

**Relevant documentation as reference**

**Definition of Ready**

- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if applicable)
- [ ] Design / Solution is accepted by Product Owner (if applicable)
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by Dianadec** [comment]
@danieltruong to add to dev checklist as needed

**Comment by LindsayMacfarlane** [comment]
Thanks @danieltruong! Assuming there is nothing for me to check/review specifically as part of this ticket? Did the investigation show if we are able to use submodules to share code between two projects?

**Comment by danieltruong** [comment]
@LindsayMacfarlane After some review, we have determined it is possible but it would take a large amount of work that is not feasible at the moment. Moving forwards we will continue sharing code in the same was as the other three projects. If we have time at a later date it would worth looking into this again.

**For future reference for future devs:**
It is possible to use GH Submodules to pull in shared code into a repo. At the moment, we know how to do that so long as the component or service does not have any dependencies. At the moment, all our shared code candidates have dependencies that also exist within the shared code library.
 
In the case of our spike:

Our shared library had:
ConfigService
CognitoAuthService

CognitoAuthService requires ConfigService to operate.

The issue exists when we imported these two services into our reserve-rec repo. When initializing CognitoAuthService, it was unable to find ConfigService despite us importing and initializing the code from the shared library.

To rectify this, we will have to further look into how to set up dependencies when creating Angular Libraries, specifically how we do it in Angular 17+.
